snmp/datadog_checks/snmp/metrics.py

```diff
@@ -44,19 +46,21 @@ def as_metric_with_inferred_type(value):
 
 def as_metric_with_forced_type(value, forced_type):
     # type: (Any, str) -> Optional[MetricDefinition]
+    value = float(sanitize_varbind_value(value))
+
     if forced_type == 'gauge':
-        return {'type': 'gauge', 'value': int(value)}
+        return {'type': 'gauge', 'value': value}
```

Comment:
@therve @florimondmanca Do you see any issue here to cast `float` instead of `int` ?

Reply:
I think it's fine. I'm not sure SNMP sends float values at all (e.g. remember how a `sysUpTimeInstance` of 1.5seconds is sent as 150 (100ths of a second), not 1.5?) which I suppose is mostly why we only sent ints before. Have you seen a use case for collecting floats?